v3

Defelo opened the pull request.

Paul-Brauckmann commented on morpheushelper/cogs/custom/__init__.py:
```diff
@@ -0,0 +1,4 @@
+from . import translations as _
```
do you need this import there?

Defelo replied:
yes, this is needed to load the translation namespaces

Paul-Brauckmann commented on morpheushelper/morpheushelper.py:
```diff
@@ -1,360 +1,36 @@
+    setup_sentry(SENTRY_DSN, Config.NAME, Config.VERSION)
 
-bot.run(os.environ["TOKEN"])
+__import__("bot").run()
```
How about if u make the import at the top level and execute it at the end

Defelo replied:
this is not possible because the config needs to be loaded before `bot` is imported

Commit:
Updated submodules

Commit:
Updated .flake8

Commit:
Updated .flake8

Commit:
Added DISABLE_PAGINATION env var

Commit:
Updated cog library

Commit:
Updated library to aa31a22 to test new send_long_embed implementation

Commit:
Improved verification log

Commit:
Added repeat_footer enforcement when using pagination

Commit:
Updated library

Commit:
Fixed on_command_error

Commit:
Fixed on_command_error

Commit:
Moved codeblock and metaquestion info commands into custom commands

Commit:
Updated cog library

Commit:
Renamed user info permission

Commit:
Added custom commands cog

Commit:
Added pagination env vars

Commit:
Updated submodules

Commit:
Merge template into develop

Commit:
Fixed library imports

Commit:
Fixed library imports

Commit:
Made github id optional for contributors

Commit:
Made github id optional for contributors

Commit:
Updated library

Commit:
Fixed sudo cog

Commit:
Refactored heartbeat cog

Commit:
Updated submodules

Commit:
Updated submodules

Commit:
Merge template into develop

Commit:
Updated submodules

Commit:
Added artifact retention to ci

Commit:
Merge template into develop

florian-sabonchi committed:
first draft with review for user_notes

dependabot[bot] committed:
Bump docker/setup-qemu-action from 1.1.0 to 1.2.0

Bumps [docker/setup-qemu-action](https://github.com/docker/setup-qemu-action) from 1.1.0 to 1.2.0.
- [Release notes](https://github.com/docker/setup-qemu-action/releases)
- [Commits](https://github.com/docker/setup-qemu-action/compare/v1.1.0...v1.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump docker/build-push-action from 2.4.0 to 2.5.0

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 2.4.0 to 2.5.0.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/v2.4.0...v2.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/cache from 2.1.5 to 2.1.6

Bumps [actions/cache](https://github.com/actions/cache) from 2.1.5 to 2.1.6.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2.1.5...v2.1.6)

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Updated dependencies

Commit:
Updated submodules

Commit:
Fixed status description in heartbeat cog

Commit:
Fixed dynamic voice channels

Commit:
Added custom_commands volume to docker-compose.yml

Commit:
Added custom_commands volume to docker-compose.yml

Commit:
Added custom command permissions

Commit:
Added suggest_role_color command

Commit:
Added optional confirmation requirement to custom commands

Commit:
Renamed morpheushelper directory to bot to avoid merge conflicts

Commit:
Fixed sf command

Commit:
Merge template into develop

Commit:
Updated dependencies and submodules

Commit:
Merge template into develop

Commit:
Merge dependabot/github_actions/actions/cache-2.1.6 into template

Commit:
Merge dependabot/github_actions/docker/setup-qemu-action-1.2.0 into template

Commit:
Merge dependabot/github_actions/docker/build-push-action-2.5.0 into template

Commit:
Merge template into develop

Commit:
Improved inactive command

Commit:
Improved inactive command

Commit:
Fixed custom command paths

Commit:
Added permission_error event handler

Commit:
Added permission_error event handler

Commit:
Added permission checks

Commit:
Added permission checks

Commit:
Added new dynamic voice channels

Commit:
Changed emoji for hidden channels

Commit:
Fixed vc show command

Commit:
Improved vc info command

Commit:
Improved vc d embed

Commit:
Added vc help command

Commit:
Improved voice handler delays

Commit:
Merge template into develop

Commit:
Updated cog library

Commit:
Improved bot info embed

Commit:
Merge develop into feature/voice-channels

Commit:
Fixed quick rejoin after kick issue

Commit:
Added reactions to bot messages

Commit:
Fixed role links in dynamic voice channels

Commit:
Fixed text channel permissions

Commit:
Fixed vc remove command

Commit:
Fixed member order in info embed

Commit:
Updated submodules

Commit:
Merge template into develop

Commit:
Fixed contributor info command

Commit:
Fixed contributor info command

Commit:
Merge develop into feature/voice-channels

Commit:
Merge develop into feature/user-notes

Commit:
Fixed and improved user notes

Commit:
Merge feature/user-notes into develop

Commit:
Merge develop into feature/voice-channels

Commit:
Added vc rename command

Commit:
Recreate channels in member_leave event

Commit:
Fixed contributor command

Commit:
Merge develop into feature/voice-channels

Commit:
Updated submodules

Commit:
Merge template into develop

Commit:
Added status icon to roles list command

Commit:
Added status icon to inactive command

Commit:
Added message clear command

Commit:
Merge develop into feature/voice-channels

dependabot[bot] committed:
Bump sqlalchemy from 1.4.17 to 1.4.18 (#220)

Bumps [sqlalchemy](https://github.com/sqlalchemy/sqlalchemy) from 1.4.17 to 1.4.18.
- [Release notes](https://github.com/sqlalchemy/sqlalchemy/releases)
- [Changelog](https://github.com/sqlalchemy/sqlalchemy/blob/master/CHANGES)
- [Commits](https://github.com/sqlalchemy/sqlalchemy/commits)

---
updated-dependencies:
- dependency-name: sqlalchemy
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump sqlalchemy from 1.4.17 to 1.4.18 (#220)

Bumps [sqlalchemy](https://github.com/sqlalchemy/sqlalchemy) from 1.4.17 to 1.4.18.
- [Release notes](https://github.com/sqlalchemy/sqlalchemy/releases)
- [Changelog](https://github.com/sqlalchemy/sqlalchemy/blob/master/CHANGES)
- [Commits](https://github.com/sqlalchemy/sqlalchemy/commits)

---
updated-dependencies:
- dependency-name: sqlalchemy
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Updated submodules

Commit:
Merge develop into feature/voice-channels

Commit:
Fixed permission error in leave event handler

Commit:
Fixed hide/show reactions

Commit:
Fixed permission errors (again)

Commit:
Fixed connect permission overwrites

Commit:
Added intel codenames

Commit:
Added RemindMe

Commit:
Merge develop into feature/voice-channels

Commit:
Added attachments to RemindMe

Commit:
Added delta to list of contributors

Commit:
Merge develop into feature/voice-channels

Commit:
Improved permission checks for role assignment commands

Commit:
Added language and version to run code output embed

Commit:
Merge develop into feature/voice-channels

Commit:
Added perma roles

Commit:
Merge develop into feature/voice-channels

Commit:
Fixed roles perma_list command

Commit:
Merge develop into feature/voice-channels

Commit:
Added discord bot token info command

Commit:
Fixed Logger Edit Distance Bug

Commit:
Improved voice role links alerts

Commit:
Updated voice channel cog

> Try to avoid duplicate channel names
> Added permission alerts
> Fixed reaction race condition
> Improved help embed

Commit:
Added discord bot token deleter

Commit:
Merge develop into feature/voice-channels

Commit:
Added merge me workflow

dependabot[bot] committed:
Bump bot/library from `b6e4679` to `373b813`

Commit:
Enabled merge me for manual changes

dependabot[bot] committed:
Merge branch 'develop' into dependabot/submodules/bot/cogs/library-c9…

dependabot[bot] committed:
Bump wemake-python-styleguide from 0.15.2 to 0.15.3

dependabot[bot] committed:
Merge develop into dependabot/github_actions/wemake-services/wemake-p…

dependabot[bot] committed:
Bump bot/cogs/library from `c9b7c0a` to `cc81bfb`

Commit:
Added codeowners

Commit:
Updated library submodule

Commit:
Merge template into develop

Commit:
Merge develop into feature/voice-channels

Commit:
Updated cogs

dependabot[bot] committed:
Bump docker/login-action from 1.9.0 to 1.10.0

dependabot[bot] committed:
Merge branch 'develop' into dependabot/pip/sqlalchemy-1.4.19

dependabot[bot] committed:
Bump bot/cogs/library from `a1c664b` to `8ead902`

Commit:
Merge template into develop

Commit:
Merge develop into feature/voice-channels

Commit:
Added cog documentation to readme

dependabot[bot] committed:
Bump bot/cogs/library from `8ead902` to `81ca91e`

Commit:
Improved ci

Commit:
Improved ci

Commit:
Updated Dockerfile

Commit:
Merge template into develop

Commit:
Merge template into develop

Commit:
Merge develop into feature/voice-channels

dependabot[bot] committed:
Bump bot/library from `0f1dabe` to `668886c`

Commit:
Merge feature/voice-channels into develop

Commit:
Added docker clean workflow

Commit:
Added docker clean workflow

dependabot[bot] committed:
Bump docker/setup-buildx-action from 1.3.0 to 1.4.1

dependabot[bot] committed:
Bump python from 3.9-alpine to 3.9.6-alpine

dependabot[bot] committed:
Bump sqlalchemy from 1.4.19 to 1.4.20

Commit:
Added S311 to .flake8

dependabot[bot] committed:
Merge branch 'develop' into dependabot/submodules/bot/cogs/library-81…

Commit:
Merge template into develop

dependabot[bot] committed:
Bump bot/library from `668886c` to `4bce8de`

Commit:
Merge template into develop

dependabot[bot] committed:
Bump docker/build-push-action from 2.5.0 to 2.6.1

Commit:
Merge template into develop

Commit:
Added random vc name list mechanism

dependabot[bot] committed:
Bump docker/setup-buildx-action from 1.4.1 to 1.5.0

dependabot[bot] committed:
Bump bot/cogs/library from `81a965d` to `23fab73`

dependabot[bot] committed:
Bump bot/cogs/library from `23fab73` to `ab8cd7a`

Commit:
Merge template into develop

dependabot[bot] committed:
Bump bot/library from `4bce8de` to `017b7aa`

dependabot[bot] committed:
Bump bot/cogs/library from `ab8cd7a` to `9bea3ec`

Commit:
Merge template into develop

Commit:
Fixed and improved mediaonly cog

Commit:
Fixed verify command

Commit:
Added documentation to cogs command

Commit:
Added cog documentation to help command

Commit:
Added documentation for suggest_role_color command

Commit:
Added encode command

Commit:
Improved suggest_role_color command

dependabot[bot] committed:
Bump bot/cogs/library from `9bea3ec` to `53c75c1`

dependabot[bot] committed:
Merge branch 'develop' into dependabot/submodules/bot/library-82833a9

Commit:
Added AutoClearCog

dependabot[bot] committed:
Bump bot/cogs/library from `53c75c1` to `859f3e8`

Commit:
Fixed vc permission overwrite bug

dependabot[bot] committed:
Bump bot/cogs/library from `859f3e8` to `dbdfbb5`

Commit:
Added emojis to command list in help embed

dependabot[bot] committed:
Bump docker/setup-buildx-action from 1.5.0 to 1.5.1

dependabot[bot] committed:
Bump bot/library from `82833a9` to `25217af`

Commit:
Added new custom commands

Commit:
Added new custom commands

Commit:
Merge template into develop

dependabot[bot] committed:
Bump bot/cogs/library from `379b88d` to `4856148`

dependabot[bot] committed:
Merge branch 'develop' into dependabot/submodules/bot/library-ca6eb6a

Commit:
Merge template into develop

Commit:
Fixed cc empty content error

Commit:
Fixed cc embed errors

dependabot[bot] committed:
Bump bot/cogs/library from `4856148` to `2934634`

Commit:
Fixed cc empty embed error

dependabot[bot] committed:
Bump bot/cogs/library from `2934634` to `6ce3982`

dependabot[bot] committed:
Merge branch 'develop' into dependabot/pip/requests-2.26.0

dependabot[bot] committed:
Merge branch 'develop' into dependabot/pip/python-dateutil-2.8.2

Commit:
Fixed and improved custom commands

  > Improved cc test warnings
  > Added ask_cc_test
  > Added order_by to cc aliases
  > Added cc alias limit
  > Added cc not found error to converter
  > Use requests instead of aiohttp in load_discohook because of a bug in aiohttp
  > Improved cc edit data cmd description
  > Added cc warnings in test mode
  > Replaced cc add enabled parameter with public parameter

dependabot[bot] committed:
Bump bot/cogs/library from `6ce3982` to `11a6b7a`

Commit:
Merge template into develop

dependabot[bot] committed:
Bump bot/library from `ca6eb6a` to `4cada2d`

dependabot[bot] committed:
Merge branch 'develop' into dependabot/pip/sqlalchemy-1.4.21

Commit:
Merge template into develop

Commit:
Updated cogs

  > Improved invite detection
  > Added poll options char limit
  > Added guild_only decorator to custom commands

dependabot[bot] committed:
Bump bot/cogs/library from `11a6b7a` to `73cdd0e`

Commit:
Improved verify too_soon error message

Commit:
Improved poll commands

Commit:
Fixed cc on_ready

Commit:
Added cc permission alerts

Commit:
Added warning to cc channel parameter

Commit:
Added permission check to cc e channel command

Commit:
Updated cogs

  > Made reports visible to team members only (fixes #31)
  > Added illegal invite posts to ulog (fixes #25)
  > Added mediaonly deletions to ulog (fixes #67)

Commit:
Updated cogs

  > Fixed #72
  > Added remove_reactions parameter to rr delete (fixes #69)
  > Added rr reinit command (fixes #68)

dependabot[bot] committed:
Bump bot/cogs/library from `73cdd0e` to `20e4ac8`

dependabot[bot] committed:
Bump bot/library from `4cada2d` to `8d10096` (#106)

Bumps [bot/library](https://github.com/PyDrocsid/library) from `4cada2d` to `8d10096`.
- [Release notes](https://github.com/PyDrocsid/library/releases)
- [Commits](https://github.com/PyDrocsid/library/compare/4cada2db993c94b34dc49936c834187473b2f317...8d10096987754e5a7c9fa2a93dc2840a92683f25)

---
updated-dependencies:
- dependency-name: bot/library
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Felix <41747605+Defelo@users.noreply.github.com>

Commit:
Merge template into develop

dependabot[bot] committed:
Bump sqlalchemy from 1.4.21 to 1.4.22

Commit:
Fixed roles config commands

dependabot[bot] committed:
Bump bot/cogs/library from `20e4ac8` to `c1e682e`

Defelo committed:
Added Discord badge to readme

Defelo committed:
Added Discord badge to readme

dependabot[bot] committed:
Bump bot/library from `8d10096` to `4abe228`

dependabot[bot] committed:
Bump bot/library from `4abe228` to `d0fcf1a`

dependabot[bot] committed:
Bump bot/library from `d0fcf1a` to `dd46143`

Commit:
Merge template into develop

Commit:
Added default permission levels to config (fixes #246)

dependabot[bot] committed:
Bump sqlalchemy from 1.4.22 to 1.4.23

Commit:
Updated cogs

  > Added missing optional permission hints
  > Fixed permissions
  > Set inline property of permission fields to false

Commit:
Updated cogs

  > Fixed inactive_duration limits
  > Added edit distance limit to btp
  > Fixed permission error in instantkick

dependabot[bot] committed:
Bump docker/build-push-action from 2.6.1 to 2.7.0

dependabot[bot] committed:
Bump bot/cogs/library from `c1e682e` to `2190795`

Commit:
Merge template into develop

Commit:
Fixed internal errors in reactionrole

Commit:
Fixed DurationConverter

dependabot[bot] committed:
Bump bot/cogs/library from `2190795` to `8f3f670`

Commit:
Removed library submodule

Commit:
Updated dependencies

Commit:
Merge template into develop

Commit:
Fixed PollOption emoji parser

Commit:
Fixed rr add command

Commit:
Fixed reddit translations

Commit:
Fixed inactivity scan command

Commit:
Fixed inactive duration limits

dependabot[bot] committed:
Bump bot/cogs/library from `8f3f670` to `9352992`

Commit:
Updated cogs

  > Fixed mod permission checks
  > Added permission checks to mod tools
  > Renamed BeTheProfessional cog
  > Removed backtick from list of allowed topic characters
  > Added limit to length of topic names

Commit:
Updated cogs

Commit:
Updated Pipfile

Commit:
Merge template into develop

TheCataliasTNT2k reviewed (approved):
Yes, it is completed.... It was a loooong way....
